### DISCUSSION TITLE
MolsMatrixToGridImage: Display legend even if molecule is None

### DISCUSSION THREAD

**Discussion by metma99** [rdkit/rdkit#7678] [Ideas] [answered]
Hello:
I am facing the following situation:
I have a matrix of molecule cores in the first row and molecule r rgoups in the first column, element (0,0) is also None.
The other matrix elements have the molecule None as the first row and the first column define the molecule.
Next, I would like to display some data only in the matrix elements where there is no molecule defined.

This seems not possible currently.
I tested by adding a molecule to one of the empty matrix elements and then the legend text appears.
As soon as the molecule is None the legend text disappears.

Am I missing something?

Any input is appreciated,
Markus

**Comment by bp-kelley** [comment, marked as answer]
MolsToGridImage doesn't render None or empty molecules.  You could simply substitute the empty molecule with a single dummy atom:

```
Draw.MolsToGridImage(
             [m if m else Chem.MolFromSmiles("*") for m in mols],
             legends=["a", "b", "c"])
```
![image](https://github.com/user-attachments/assets/5478c354-60d9-4b53-83ec-509c625e3969)

**Comment by bertiewooster** [comment]
As @bp-kelley implies, [MolsMatrixToGridImage](https://www.rdkit.org/docs/source/rdkit.Chem.Draw.html#rdkit.Chem.Draw.MolsMatrixToGridImage) calls [MolsToGridImage](https://www.rdkit.org/docs/source/rdkit.Chem.Draw.html#rdkit.Chem.Draw.MolsToGridImage), so they will render the same way for a None or empty molecule. Nice trick to make an empty molecule show up as `*`! There have been times I've wanted to have a legend below a blank cell.

@metma99 Glad you're finding MolsMatrixToGridImage, which I contributed, useful! Did you publish this work online?
